### ISSUE TITLE
Nextcloud 12 compatibility

### ISSUE THREAD

**Issue by jasonbayton** [pellaeon/registration#75]
Hi, 

When clicking on the registration button since updating to NC 12, the app reports an internal error:

![image](https://user-images.githubusercontent.com/3178412/27006517-800fe5c6-4e2d-11e7-9df4-5c2c766164b6.png)

Log as follows:
 
```
Error: Call to undefined method OC\Server::getDb()
/var/www/html/3rdparty/pimple/pimple/src/Pimple/Container.php - line 113: OCA\Registration\AppInfo\Application->OCA\Registration\AppInfo\{closure}(Object(OC\AppFramework\DependencyInjection\DIContainer))
/var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php - line 115: Pimple\Container->offsetGet('PendingRegist')
/var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php - line 423: OC\AppFramework\Utility\SimpleContainer->query('PendingRegist')
/var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php - line 408: OC\AppFramework\DependencyInjection\DIContainer->queryNoFallback('PendingRegist')
/var/www/html/apps/registration/appinfo/application.php - line 41: OC\AppFramework\DependencyInjection\DIContainer->query('PendingRegist')
/var/www/html/3rdparty/pimple/pimple/src/Pimple/Container.php - line 113: OCA\Registration\AppInfo\Application->OCA\Registration\AppInfo\{closure}(Object(OC\AppFramework\DependencyInjection\DIContainer))
/var/www/html/lib/private/AppFramework/Utility/SimpleContainer.php - line 115: Pimple\Container->offsetGet('RegisterControl...')
/var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php - line 423: OC\AppFramework\Utility\SimpleContainer->query('RegisterControl...')
/var/www/html/lib/private/AppFramework/DependencyInjection/DIContainer.php - line 408: OC\AppFramework\DependencyInjection\DIContainer->queryNoFallback('RegisterControl...')
/var/www/html/lib/private/AppFramework/App.php - line 91: OC\AppFramework\DependencyInjection\DIContainer->query('RegisterControl...')
/var/www/html/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47: OC\AppFramework\App main('RegisterControl...', 'askEmail', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)
[internal function] OC\AppFramework\Routing\RouteActionHandler->__invoke(Array)
/var/www/html/lib/private/Route/Router.php - line 299: call_user_func(Object(OC\AppFramework\Routing\RouteActionHandler), Array)
/var/www/html/lib/base.php - line 1000: OC\Route\Router->match('/apps/registrat...')
/var/www/html/index.php - line 40: OC handleRequest()
{main}
```
A similar issue was reported on the weather app, now resolved: 
https://github.com/nextcloud/weather/commit/8f41dd3fd0ad824038f907a4ad51417d56fcf157

And the topic on NC if you need it:
https://help.nextcloud.com/t/registration-in-nc12-internal-error/13897

Thanks!

**Comment by HelFANS** [comment]
We have the same error. Please check the problem and make a new version available.
We get the message in logging:
Error: Call to undefined method OC\Server::getDb()

**Comment by Tarry91** [comment]
I do have the same issue.. Is there a fix yet?

**Comment by jasonbayton** [comment]
No acknowledgement at all :(

**Comment by Tarry91** [comment]
Mhh... https://github.com/pellaeon/registration/commit/324c218c8d4d36ebea6b3cc68c344456f4187abd

There seems to be a fix.

**Comment by jasonbayton** [comment]
Perfect nice catch there! I'll test this later and close this off when confirmed.

**Comment by Tarry91** [comment]
I did download the whole master branch and replaced the complete app/registration folder with it. It works fine and as intended for me.

**Comment by pellaeon** [comment]
@Tarry91 Thanks for the testing, I released v0.2.5 to solve this.
